client/implementation/src/main/java/io/smallrye/graphql/client/impl/typesafe/json/JsonReader.java

```diff
@@ -22,16 +22,25 @@
 import io.smallrye.graphql.client.impl.typesafe.reflection.FieldInfo;
 import io.smallrye.graphql.client.impl.typesafe.reflection.TypeInfo;
 import io.smallrye.graphql.client.typesafe.api.ErrorOr;
+import io.smallrye.graphql.client.typesafe.api.TypesafeClientResponse;
 
 public class JsonReader extends Reader<JsonValue> {
     public static Object readJson(String description, TypeInfo type, JsonValue value, FieldInfo field) {
         return readJson(new Location(type, description), type, value, field);
     }
 
+    public static Object readJsonTypesafeClientResponse(String description, TypeInfo type, JsonValue value, FieldInfo field) {
+        return readJsonTypesafeClientResponse(new Location(type, description), type, value, field);
+    }
+
```

Comment:
We're just trying to go with the same approach as `readJson`, where there's one public method calling one private method, to make it look consistent, even though it might mean one extra method call.